test/lib/Assertion.test.ts

```diff
@@ -263,6 +276,40 @@ describe("[Unit] Assertion.test.ts", () => {
     });
   });
 
+  describe(".toBeInstanceOf", () => {
+    context("when the value is an instance of the constructor", () => {
+      const variants = [
+        [new Date(), Date],
+        [new Error("failed!"), Error],
+        [new Car("Pontiac GT-37"), Car]
+      ] as const;
+
+      variants.forEach(([value, Constructor]) => {
+        it(`[Instance: ${Constructor.name}]: returns the assertion instance`, () => {
+          const test = new Assertion(value);
+
+          assert.deepStrictEqual(test.toBeInstanceOf(Constructor), test);
+          assert.throws(() => test.not.toBeInstanceOf(Constructor), {
+            message: `Expected value NOT to be an instance of <${Constructor.name}>`,
+            name: AssertionError.name
+          });
+        });
+      });
+    });
+
+    context("when the vlaue in not an instance of the constructor", () => {
+      it("throws an assertion error", () => {
+        const test = new Assertion(new Date());
+
+        assert.throws(() => test.toBeInstanceOf(Car), {
+          message: "Expected value to be an instance of <Car>",
```

Comment:
@JoseLion I think it would be better to not hard-coding the class name.
```suggestion
          message: `Expected value to be an instance of <${Car.name}>`,
```

Reply:
In the case of tests, it's ok to hard code it. We want to be sure of the expected result of the message